### PR TITLE
feat: Add Rockset as an OnlineStore

### PR DESCRIPTION
**Pull request by danielin917**
This commit adds Rockset as a contributed online store. It implements the Update, Teardown, Read and Write apis using the Rockset python client.

Signed-off-by: Daniel Lin <dan@rockset.com>

**What this PR does / why we need it**:
This PR adds Rockset as an implemented OnlineFeatureStore

**Which issue(s) this PR fixes**:

Fixes #
N/A

### CODE REVIEW

**Comment by danielin917** [comment]
Hello! Integrations tests have all passed for the online store. Did not add a local universal test to the Makefile since there is no docker image to use.

**Comment by danielin917** [comment]
👋 Just checking in to see if there is anything I should do to make this easier to review

**Comment by mavysavydav** [comment]
hi! Do you know who might have context on these changes who may be better suited to review? I also recently stepped down from my role as a maintainer so am not as involved in reviewing these for the time being. Apologies for the delay! If there's nobody you have in mind who might be a good fit for reviewing, I can pass this on to someone else in the maintainer team.

**Comment by danielin917** [comment]
Not sure who the best person is, I think @felixwang9817 mentioned he was interested but not sure who else could review

**Comment by mavysavydav** [comment]
@felixwang9817 -- were you interested in taking a look at this or do you know who might be?

File: sdk/python/requirements/py3.9-ci-requirements.txt
Comment (adchia) [inline sdk/python/requirements/py3.9-ci-requirements.txt:2]: seems like you generated this with the wrong python version?

File: docs/SUMMARY.md
Comment (felixwang9817) [inline docs/SUMMARY.md:92]: this line can be removed - we only need the `[Rockset (contrib)]` line!

File: docs/reference/online-stores/README.md
Comment (felixwang9817) [inline docs/reference/online-stores/README.md:17]: similar to `SUMMARY.md`, this block can be removed

File: docs/reference/online-stores/rockset.md
Comment (felixwang9817) [inline docs/reference/online-stores/rockset.md:1]: would make this `Rockset (contrib)`

File: docs/reference/online-stores/rockset.md
Comment (felixwang9817) [inline docs/reference/online-stores/rockset.md:5]: this is formatted strangely, would probably turn this into a sentence

File: docs/reference/online-stores/rockset.md
Comment (felixwang9817) [inline docs/reference/online-stores/rockset.md:9]: it's -> its

inerts -> inserts

File: docs/reference/online-stores/rockset.md
Comment (felixwang9817) [inline docs/reference/online-stores/rockset.md:13]: this section is formatted strangely, might turn this into a code block?

File: sdk/python/requirements/py3.8-ci-requirements.txt
Comment (felixwang9817) [inline sdk/python/requirements/py3.8-ci-requirements.txt:2]: you should set up a virtualenv with python 3.8 and actually run this command with python 3.8

File: setup.py
Comment (felixwang9817) [inline setup.py:203]: you should also add this down below to `extras_require`

this will allow people to install Feast with Rockset by doing `pip install feast[rockset]`

**Review by felixwang9817 — COMMENTED** [review]
looks good overall! just a few nits

**Comment by felixwang9817** [comment]
hey @danielin917 thanks for the PR and sorry for the delay on the review - I'm happy to help you get this PR merged

I left a few nits above, and I think you'll want to rebase to ensure you can pass the latest tests!

File: sdk/python/requirements/py3.8-ci-requirements.txt
Comment (danielin917) [reply sdk/python/requirements/py3.8-ci-requirements.txt]: Yup all done, I naively thought updating the local env var would do everything

**Comment by danielin917** [comment]
Thanks for all the comments!

File: sdk/python/requirements/py3.9-ci-requirements.txt
Comment (danielin917) [reply sdk/python/requirements/py3.9-ci-requirements.txt]: Yea, thanks for the catch!

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by felixwang9817** [comment]
@danielin917 I stamped but looks like there's actually a few outstanding errors that need to be addressed

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by felixwang9817** [comment]
btw @danielin917 for the DCO check you'll need to make sure all your commits are signed off properly!

**Comment by danielin917** [comment]
@felixwang9817 A few changes since last lgtm to get everything working.
- The biggest is that I commented out Rockset from the integration repo config. Until we have a dedicated instance to run CI for this like https://github.com/feast-dev/feast/pull/2258/files we will not have an account to test against (not sure if this can be provided for free in the near future).
- Next I pinned mypy to 0.98.1 since the upgrade manifested a bunch of unrelated lints that might be better handled in a separate diff
- Also, I added an explicit dependency on httpx as it was recently changed for the Starlette dependency which is needed by fastapi and there is currently no great way to pull through fastapi itself: https://github.com/tiangolo/fastapi/issues/5656
- Finally a few lint edits for the new mypy version that I added before pinning due to the overwhelming number of changes needed in other files. (Rockset integration tests still passed when run against my account)

**Comment by adchia** [comment]
Still seeing a lot of lint errors?

Screenshot 2023-01-23 at 1 15 35 PM

**Comment by danielin917** [comment]
> Still seeing a lot of lint errors?
> 
> Screenshot 2023-01-23 at 1 15 35 PM
Sorry for the back and forth. pylint errors fixed up and locally runs clean now. I had to pin types-protobuf to get rid of the lint issues introduced in 4+ lots of:
```
Library stubs not installed for "google.*"
```

**Comment by adchia** [comment]
This seems fine. We'll probably have to revisit once libraries like tensorflow force protobuf v4

**Comment by danielin917** [comment]
Hmm looks like bigquery test failed with rate limit error. Let mee see if I can retrigger

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by danielin917** [comment]
ERROR: Cannot uninstall types-protobuf 4.21.0.3, RECORD file not found. You might be able to recover from this via: 'pip install --force-reinstall --no-deps types-protobuf==4.21.0.3'.

Not sure if there is a way to just rerun this test and this goes away or might need to play with dependancies more.

**Comment by felixwang9817** [comment]
> ERROR: Cannot uninstall types-protobuf 4.21.0.3, RECORD file not found. You might be able to recover from this via: 'pip install --force-reinstall --no-deps types-protobuf==4.21.0.3'.
> 
> Not sure if there is a way to just rerun this test and this goes away or might need to play with dependancies more.

@danielin917 looks like it's probably just a dependencies issue - I'll look into it today/tmrw!

File: setup.py
Comment (felixwang9817) [inline setup.py:159]: @danielin917 why is this necessary?

File: setup.py
Comment (felixwang9817) [inline setup.py:182]: @danielin917 why is this necessary? the error message in the failing unit tests is `ERROR: Cannot uninstall types-protobuf 4.21.0.3, RECORD file not found. You might be able to recover from this via: 'pip install --force-reinstall --no-deps types-protobuf==4.21.0.3'.`, so this line seems very suspicious to me

Comment (danielin917) [reply setup.py:182]: Hey yea, mentioned above that I pinned this type because the linter was producing a lot of errors using the new version `Library stubs not installed for "google.*"`. One thing that might be causing this is that we have:
 "mypy-protobuf==3.1",

**Comment by danielin917** [comment]
Not sure what happened between earlier testing and now but after removing cap on mypy and types-proto and this time I saw no lint errors locally

File: setup.py
Comment (danielin917) [reply setup.py]: I was seeing a lot of unrelated lint errors before from different files. But actually now after removing the cap lint passed so not sure what changed

**Comment by danielin917** [comment]
Ugh, I am going to open a separate commit for upgrading mypy and deal with all of the linting issues there to unblock this diff as is instead of playing around with package dependencies

**Comment by danielin917** [comment]
Sorry to keep pinging just want to test the workflow after the rebase

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by felixwang9817** [comment]
@danielin917 we finally got this merged in!! thanks a ton for the contribution, and for your patience throughout the review process - I know we ran into a bunch of issues throughout 😅

**Comment by danielin917** [comment]
Ahh sweet! Thanks @felixwang9817 and @adchia for the assist!

**Pull request merged**
